various-receipes/nestjs/app/test/basic-tests.test.ts

```diff
@@ -0,0 +1,61 @@
+import { INestApplication } from '@nestjs/common';
+import { Test } from '@nestjs/testing';
+import { AppModule } from '../app.module';
+const request = require('supertest');
+const sinon = require('sinon');
+const nock = require('nock');
+
+let app: INestApplication;
+
+beforeAll(async (done) => {
+  // ️️️✅ Best Practice: Place the backend under test within the same process
+  const moduleFixture = await Test.createTestingModule({
```

Comment:
Major - That's a design decision that deviates from our main example where the production code exposes two methods: start/stop and the testing code is not even being aware of the interface of internals objects (INestApp). I'm not saying that what you did here is inferior, but it worth a discussion.

Reasons to prefer start/stop:
1. The production code encapsulates how it starts and stops. The test code doesn't care
2. There can't be deviation between how you define the module in production vs in-testing. Here you can configure your INestApp differently
3. We move to axios so there is no need to know about the Nest internals, just get back a URL

On the other hand, what you aligned with Nest documentation. But do they really understand what they are doing?